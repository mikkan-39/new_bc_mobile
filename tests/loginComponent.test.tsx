import * as React from "react";
import LoginComponent from "../src/components/loginComponent";
import { render, screen, fireEvent } from "./myTestRenderer";
import reactotron from "reactotron-react-native";

describe("loginComponent test", () => {
  reactotron.log = () => {};

  it("should match snapshot", () => {
    render(<LoginComponent loginCallback={() => {}} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("should render username", async () => {
    const expectedUsername = "123";
    render(<LoginComponent loginCallback={() => {}} />);
    fireEvent.changeText(screen.getByTestId("username"), expectedUsername);
    expect(screen.getByDisplayValue(expectedUsername)).toBeTruthy();
  });

  it("should render password", async () => {
    const expectedPassword = "321";
    render(<LoginComponent loginCallback={() => {}} />);
    fireEvent.changeText(screen.getByTestId("password"), expectedPassword);
    expect(screen.getByDisplayValue(expectedPassword)).toBeTruthy();
  });

  it("should pass creds", async () => {
    const mockFn = jest.fn();
    const expectedUsername = "123";
    const expectedPassword = "321";

    render(<LoginComponent loginCallback={mockFn} />);

    fireEvent.changeText(screen.getByTestId("username"), expectedUsername);
    fireEvent.changeText(screen.getByTestId("password"), expectedPassword);
    fireEvent.press(screen.getByTestId("loginButton"));

    expect(mockFn).toBeCalledWith({
      username: expectedUsername,
      password: expectedPassword,
    });
  });
});
