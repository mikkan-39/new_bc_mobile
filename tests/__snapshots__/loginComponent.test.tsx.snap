// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginComponent test should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#eee",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-start",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "flexShrink": 3,
          "height": "15%",
          "width": 10,
        }
      }
    />
    <View
      style={
        {
          "aspectRatio": 1,
          "backgroundColor": "#fff",
          "borderRadius": 30,
          "elevation": 13,
          "flexShrink": 1,
          "height": undefined,
          "maxWidth": 300,
          "minHeight": 100,
          "minWidth": 100,
          "padding": 10,
          "shadowColor": "#976dd7",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.39,
          "shadowRadius": 8.3,
          "width": "50%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/logo.png",
          }
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexShrink": 5,
          "height": 25,
          "maxHeight": "15%",
          "minHeight": 0,
          "width": 10,
        }
      }
    />
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 15,
          "color": "#000",
          "elevation": 13,
          "fontSize": 20,
          "height": 50,
          "marginTop": 25,
          "maxWidth": 300,
          "padding": 10,
          "paddingHorizontal": 15,
          "shadowColor": "#976dd7",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.39,
          "shadowRadius": 8.3,
          "width": "75%",
        }
      }
      testID="username"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="go"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 15,
          "color": "#000",
          "elevation": 13,
          "fontSize": 20,
          "height": 50,
          "marginTop": 25,
          "maxWidth": 300,
          "padding": 10,
          "paddingHorizontal": 15,
          "shadowColor": "#976dd7",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.39,
          "shadowRadius": 8.3,
          "width": "75%",
        }
      }
      testID="password"
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#976dd7",
          "borderRadius": 15,
          "elevation": 24,
          "height": 50,
          "justifyContent": "center",
          "margin": 25,
          "maxWidth": 200,
          "opacity": 1,
          "shadowColor": "#976dd7",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.58,
          "shadowRadius": 16,
          "width": "50%",
        }
      }
      testID="loginButton"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 24,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
